### PR TITLE
[ONNX] Fix adding output nodes in NNCFGraph

### PR DESCRIPTION
**Pull request by kshpv**
### Changes

Do not add edges for the output node if it has no parents. E.g. output is put on the initializer.

### Reason for changes

125248

### Related tickets

125248

### Tests

Checked on reproducer in the ticket

### CODE REVIEW

**Review by alexsu52 — COMMENTED** [review]
LGTM. Please, add a test.

**Comment by kshpv** [comment]
> LGTM. Please, add a test.

Added test

File: tests/onnx/data/reference_graphs/original_nncf_graph/synthetic/output_with_no_parents_model.dot [hunk 1]
```diff
@@ -0,0 +1,8 @@
+strict digraph  {
+"0 Conv1" [id=0, type=Conv];
+"1 nncf_model_input_0" [id=1, type=nncf_model_input];
+"2 nncf_model_output_0" [id=2, type=nncf_model_output];
+"3 nncf_model_output_1" [id=3, type=nncf_model_output];
```
Comment (alexsu52) [inline tests/onnx/data/reference_graphs/original_nncf_graph/synthetic/output_with_no_parents_model.dot:5]: It looks like the graph is not connected. Let's discuss offline what impact on algorithms.

Comment (kshpv) [reply tests/onnx/data/reference_graphs/original_nncf_graph/synthetic/output_with_no_parents_model.dot]: The logic was updated. Outputs with no parent node is not added to the NNCFGraph.

**Comment by kshpv** [comment]
E2E ONNX - 654 passed
ONNX Conformance - 356 passed

**Review by alexsu52 — APPROVED** [review]
LGTM

**Pull request merged**